include/translated-term.php

```diff
@@ -340,4 +340,190 @@ protected function get_db_infos() {
+	 * @param array $args {
+	 *     Optional. Array of arguments for updating a term.
+	 *
+	 *     @type string       $alias_of     Slug of the term to make this term an alias of.
+	 *                                      Default empty string. Accepts a term slug.
+	 *     @type string       $description  The term description. Default empty string.
+	 *     @type int          $parent       The id of the parent term. Default 0.
+	 *     @type string       $slug         The term slug to use. Default empty string.
+	 *     @type PLL_Language $lang         The term language object.
+	 *     @type string[]     $translations The translation group to assign to the term with language slug as keys and `term_id` as values.
+	 * }
+	 * @return array|WP_Error An array containing the `term_id` and `term_taxonomy_id`,
+	 *                        WP_Error otherwise.                  WP_Error otherwise.
+	 */
+	public function update( int $term_id, array $args = array() ) {
+		$term = get_term( $term_id );
+		if ( ! $term instanceof WP_Term ) {
+			return new WP_Error( 'invalid_term', __( 'Empty Term.', 'polylang' ) );
+		}
+
+		/** @var PLL_Language $language */
+		$language = $this->get_language( $term_id );
+		if ( ! empty( $args['lang'] ) ) {
+			$language = $this->languages->get( $args['lang'] );
+			if ( ! $language instanceof PLL_Language ) {
+				return new WP_Error( 'invalid_language', __( 'Please provide a valid language.', 'polylang' ) );
+			}
+
+			$this->set_language( $term_id, $language );
+		}
+
+		$parent = $args['parent'] ?? $term->parent;
+		$this->toggle_inserted_term_filters( $language, $parent );
+		$term = wp_update_term( $term->term_id, $term->taxonomy, $args );
+		$this->toggle_inserted_term_filters( $language, $parent );
+
+		if ( ! empty( $args['translations'] ) ) {
+			$this->save_translations(
+				$term_id,
+				array_merge(
+					$args['translations'],
+					array(
+						$language->slug => $term_id,
+					)
+				)
+			);
+		}
+
+		return $term;
+	}
+
+	/**
+	 * Toggles Polylang term slug filters management.
+	 * Must be used before and after any term slug modification or insertion.
+	 *
+	 * @since 3.7
+	 *
+	 * @param PLL_Language $language The language to use.
+	 * @param int          $parent The parent term id to use.
+	 * @return void
+	 */
+	private function toggle_inserted_term_filters( $language, $parent ): void {
+		static $enabled = false;
+
+		if ( $enabled ) {
+			// Clean up!
+			remove_filter( 'pll_inserted_term_language', $this->get_slug_management_language_callback( $language ) );
+			remove_filter( 'pll_inserted_term_parent', $this->get_slug_management_parent_callback( $parent ) );
+			$enabled = false;
+		} else {
+			// Set term parent and language for suffixed slugs.
+			add_filter( 'pll_inserted_term_language', $this->get_slug_management_language_callback( $language ) );
+			add_filter( 'pll_inserted_term_parent', $this->get_slug_management_parent_callback( $parent ) );
+			$enabled = true;
+		}
+	}
```

Comment:
Wouldn't this be much simpler?
```suggestion
	private function toggle_inserted_term_filters( $language, $parent ): void {
		static $callback = array();

		if ( ! empty( $callback ) ) {
			// Clean up!
			remove_filter( 'pll_inserted_term_language', $callback[ $language ] );
			remove_filter( 'pll_inserted_term_parent', $callback[ $parent ] );
			$callback = array();
			return;
		}
		
		$callback[ $language ] = function () use ( $language ) {
			return $language;
		};

		$callback[ $parent ] = function () use ( $parent ) {
			return $parent;
		};

		// Set term parent and language for suffixed slugs.
		add_filter( 'pll_inserted_term_language', $callback[ $language ] );
		add_filter( 'pll_inserted_term_parent', $callback[ $parent ] );
	}
```

Reply:
Good idea, done in dc355a95.
Even though I had to adapt a bit to avoid side effects in case the method is called consecutively with different languages/parents when adding the filters. Also fixed an illegal offset type with language obects.